examples/rng.rs

```diff
@@ -8,20 +8,13 @@ use panic_semihosting as _;
 
 use cortex_m_rt::entry;
 use cortex_m_semihosting::hprintln;
-use stm32f7xx_hal::{
-    pac,
-    prelude::*,
-    rng::*,
-};
+use stm32f7xx_hal::{pac, prelude::*};
 
 #[entry]
 fn main() -> ! {
     let p = pac::Peripherals::take().unwrap();
-    let rcc = p.RCC.constrain();
-    let _clocks = rcc.cfgr.sysclk(32.mhz()).freeze();
     let mut rng = p.RNG.init();
     let val = rng.get_rand().unwrap();
     hprintln!("random value {}", val).unwrap();
-    loop{
-    }
+    loop {}
```

Comment:
Empty loops can cause undefined behavior, due to a bug in LLVM (https://github.com/rust-lang/rust/issues/28728). I believe that adding `core::sync::atomic::spin_loop_hint();` to the body of the loop is the canonical way to prevent this.

```suggestion
    loop {
        core::sync::atomic::spin_loop_hint();
    }
```

Reply:
Yeah i forgot to add it